pallets/funding/src/instantiator.rs

```diff
@@ -1876,10 +1887,11 @@ pub mod async_features {
 
 		let mut inst = instantiator.lock().await;
 
-		match inst.get_project_details(project_id).status {
-			ProjectStatus::RemainderRound | ProjectStatus::FundingSuccessful => Ok(()),
-			_ => panic!("Bad state"),
-		}
+		assert_matches!(
```

Comment:
Do we actually need this library for a single assert? Would rather remove the library and have 2-3 lines more code to do this assert manually. Cant we do something like `assert!(matches!(status, Remainder | Funding ))`

Reply:
we use it extensively in tests.rs, so why not use it here too even if we use it once?
I get that if we use it only once throughout the code base then it's silly to download a library for it, but since we already have it in the workspace why not use it and reduce code bloat?